reader.go

```diff
@@ -261,3 +261,20 @@ func (r *Reader) WriteTo(w io.Writer) (n int64, err error) {
 		}
 	}
 }
+
+// CheckLZ4 returns a bool indicating if the first 32 byets mactch LZ4 header
+// from the io.Reader stream.
+//
+// False if it could not determine, e.g. error in reading from io.Reader.
+//
+// Don't pass original io.Reader for checking, otherwise internal read pos of
+// the io.Reader will be off by 32, causing a reader error. Instead, consider
+// pre-read bytes into a slice for archive checking, and then io.MultiReader
+// to concatenate it for the real lz4 reading.
+func CheckLZ4(r io.Reader) (bool, error) {
+	f := lz4stream.NewFrame()
+	if err := f.ParseHeaders(r); err != nil {
```

Comment:
Please check for the lz4errors.ErrInvalidFrame error as well and return a nil error if it is so.

Reply:
done